File: from-2601-to-2700/2661-first-completely-painted-row-or-column/solution.go
func firstCompleteIndex(arr []int, mat [][]int) int {
    mR, mC := make(map[int]int), make(map[int]int)
    m, n := len(mat), len(mat[0])
    for r := 0; r < m; r ++ {
        for c := 0; c < n; c ++ {
            mR[mat[r][c]] = r
            mC[mat[r][c]] = c
        }
    }
    rows := make([]int, m)
    cols := make([]int, n)
    var ri, ci int
    for i := 0; i < len(arr); i ++ {
        ri = mR[arr[i]]
        rows[ri] += 1
        if rows[ri] == n {
            return i
        }
        ci = mC[arr[i]]
        cols[ci] += 1
        if cols[ci] == m {
            return i
        }
    }
    return -1
}

File: from-2601-to-2700/2661-first-completely-painted-row-or-column/solution.py
class Solution:
    def firstCompleteIndex(self, arr: List[int], mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        position_map = {mat[row][col]: (row, col) for row in range(m) for col in range(n)}
        row_count = [n] * m
        col_count = [m] * n
        for idx, val in enumerate(arr):
            row, col = position_map[val]
            row_count[row] -= 1
            col_count[col] -= 1
            if row_count[row] == 0 or col_count[col] == 0:
                return idx

        return -1